FileMapper/API_util.js

```diff
@@ -0,0 +1,307 @@
+        prev = (fldrs.hasNext()) ? fldrs.next(): null;
+      }
+
+      // Since previous folder is null that means the path is not found
+      // in drive hence it will return null
+      if (prev === null) {
+        found = true;
+      }
+    }
+
+    // If previous folder wasn't found null 
+    if (!found) {
+      found = true;
+      driveId = getDestinationId(destination, prev, isFile);
+    }
+  }
+
+  return driveId;
+}
+
+/** 
+ * Gets the destination drive id of a file/folder having same mimetype as the local file  
+ *
+ * @param {String} destination The File whose mapping is to be found
+ * @param {FolderObject} folder The Folder in which the file is to be searched
+ * @param {boolean} isFile To signify whether its a file or folder
+ * @return {String} driveId The drive id of the local file/folder if found or null if file 
+ *                  is not found
+ */
+function getDestinationId(destination, folder, isFile) {
+  var driveId = null;
+  if (isFile) {
+    var extension = getExtension(destination);
+    var mime = getMimeTypeFromExtension(extension);
+    var files = folder.getFilesByName(destination);
+    while (files.hasNext()) {
+      var x = files.next();
+      if (x.getMimeType() === mime) {
+        driveId = x.getId();
+      }
+    }
+  }
+  else {
+    var fldrs = folder.getFoldersByName(destination);
+    if (fldrs.hasNext()) {
+      var x = fldrs.next();
+      driveId = x.getId();
+    }
+  }
+
+  return driveId;
+}
+
+/**
+ * Create a new file or folder in place of a deleted one
+ * 
+ * @param {String} localPath The local destination path whose mapping was marked deleted
+ * @return {String} driveId The drive id of the newly created file or folder
+ */
+// Might need to modify this one 
+function createDeletedDestination(localPath) {
+  var documentProperties = PropertiesService.getDocumentProperties();
+  var value = documentProperties.getProperty(localPath);
+  var mapping = JSON.parse(value);
+
+  var path = mapping.drivepath;
+  // Split the drive path which is connected by forward slash (hence isUnix is set true)
+  var folders = splitPath(path, true);
+  var driveId = mapping.id;
+  if (folders.length > 0) {
+    var destinationName = folders[folders.length - 1];
+    var prev = DriveApp.getRootFolder();
+    var curr;
+    
+    for (var i = 1; i < (folders.length - 1); i++) {
+      curr = folders[i];
+      if (curr && prev) {
+        var fldrs = prev.getFoldersByName(curr);
+        prev = (fldrs.hasNext()) ? fldrs.next(): prev.createFolder(curr);
+      }
+    }
+
+    var destination;
+    if(!mapping.isfolder){
+      var extension = getExtension(localPath);
+      if (extension === ""){
+        extension = "txt";
+      }
+      var mimetype = getMimeTypeFromExtension(extension);
+      destination = prev.createFile(destinationName, "", mimetype);
+    }
+    else{
+      destination = prev.createFolder(destinationName);
+    }
+    driveId = destination.getId();
```

Comment:
I see this logic duplicated in many places, can we extract this out into a separate function?

Reply:
I changed one of the functions and made it to call the other so now there seems no duplication.